doc/optimizers.md

```diff
@@ -152,6 +152,170 @@ optimizer.Optimize(f, coordinates);
  * [Adam: A Method for Stochastic Optimization](http://arxiv.org/abs/1412.6980)
  * [Differentiable separable functions](#differentiable-separable-functions)
 
+## AdamR
+
+*An optimizer for [differentiable separable functions](#differentiable-separable-functions).*
+
+AdamR is a modified version of Adam with warm restarts. This allows the optimiser to have a automated decreasing
+step size without modification. AdamR uses the Cyclical Decay decay policy.
```

Comment:
Should we remove the second decay?

Reply:
I think we should replace it with "AdamR uses Cyclical Decay as its default decay policy."

Reply:
Sounds good.